src/main/java/com/nourish1709/hibernatesessioncash/session/action/ActionQueue.java

```diff
@@ -0,0 +1,21 @@
+package com.nourish1709.hibernatesessioncash.session.action;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class ActionQueue {
+
+    // FIFO-manner data structure that represents actions
+    private final Queue<EntityAction> entityActionQueue = new LinkedList<>();
```

Comment:
Actually ActionQueue in ORM shouldn't be represented by FIFO, rather by priorityQueue where each type of query has it's own priority and runs depending on this priority. It is needed for orm optimization and avoid sending redundant queries. You can check basic implementation of queue in our Bibernate project.